# Review diff: upstream pyrender a59963ef890891656fd17c90e12d663233dcaa99 -> this copy.
# Generated by scripts/vendor_pyrender.py. Deleted files are listed without their
# content (not meant for `git apply`). LICENSE is copied verbatim from upstream.
--- a/pyrender/__init__.py
+++ b/pyrender/__init__.py
@@ -9,7 +9,6 @@
 from .node import Node
 from .scene import Scene
 from .renderer import Renderer
-from .viewer import Viewer
 from .offscreen import OffscreenRenderer
 from .version import __version__
 from .constants import RenderFlags, TextAlign, GLTF
@@ -18,7 +17,7 @@
     'Camera', 'PerspectiveCamera', 'OrthographicCamera', 'IntrinsicsCamera',
     'Light', 'PointLight', 'DirectionalLight', 'SpotLight',
     'Sampler', 'Texture', 'Material', 'MetallicRoughnessMaterial',
-    'Primitive', 'Mesh', 'Node', 'Scene', 'Renderer', 'Viewer',
+    'Primitive', 'Mesh', 'Node', 'Scene', 'Renderer',
     'OffscreenRenderer', '__version__', 'RenderFlags', 'TextAlign',
     'GLTF'
 ]
--- a/pyrender/camera.py
+++ b/pyrender/camera.py
@@ -5,14 +5,12 @@
 """
 import abc
 import numpy as np
-import six
 import sys
 
 from .constants import DEFAULT_Z_NEAR, DEFAULT_Z_FAR
 
 
-@six.add_metaclass(abc.ABCMeta)
-class Camera(object):
+class Camera(abc.ABC):
     """Abstract base class for all cameras.
 
     Note
deleted: pyrender/font.py
deleted: pyrender/fonts/OpenSans-Bold.ttf
deleted: pyrender/fonts/OpenSans-BoldItalic.ttf
deleted: pyrender/fonts/OpenSans-ExtraBold.ttf
deleted: pyrender/fonts/OpenSans-ExtraBoldItalic.ttf
deleted: pyrender/fonts/OpenSans-Italic.ttf
deleted: pyrender/fonts/OpenSans-Light.ttf
deleted: pyrender/fonts/OpenSans-LightItalic.ttf
deleted: pyrender/fonts/OpenSans-Regular.ttf
deleted: pyrender/fonts/OpenSans-Semibold.ttf
deleted: pyrender/fonts/OpenSans-SemiboldItalic.ttf
--- a/pyrender/light.py
+++ b/pyrender/light.py
@@ -5,7 +5,6 @@
 """
 import abc
 import numpy as np
-import six
 
 from OpenGL.GL import *
 
@@ -16,8 +15,7 @@
 
 
 
-@six.add_metaclass(abc.ABCMeta)
-class Light(object):
+class Light(abc.ABC):
     """Base class for all light objects.
 
     Parameters
--- a/pyrender/material.py
+++ b/pyrender/material.py
@@ -7,15 +7,13 @@
 """
 import abc
 import numpy as np
-import six
 
 from .constants import TexFlags
 from .utils import format_color_vector, format_texture_source
 from .texture import Texture
 
 
-@six.add_metaclass(abc.ABCMeta)
-class Material(object):
+class Material(abc.ABC):
     """Base for standard glTF 2.0 materials.
 
     Parameters
--- a/pyrender/mesh.py
+++ b/pyrender/mesh.py
@@ -84,8 +84,8 @@
         """(2,3) float : The axis-aligned bounds of the mesh.
         """
         if self._bounds is None:
-            bounds = np.array([[np.infty, np.infty, np.infty],
-                               [-np.infty, -np.infty, -np.infty]])
+            bounds = np.array([[np.inf, np.inf, np.inf],
+                               [-np.inf, -np.inf, -np.inf]])
             for p in self.primitives:
                 bounds[0] = np.minimum(bounds[0], p.bounds[0])
                 bounds[1] = np.maximum(bounds[1], p.bounds[1])
--- a/pyrender/offscreen.py
+++ b/pyrender/offscreen.py
@@ -128,18 +128,18 @@
 
     def _create(self):
         if 'PYOPENGL_PLATFORM' not in os.environ:
-            from pyrender.platforms.pyglet_platform import PygletPlatform
+            from .platforms.pyglet_platform import PygletPlatform
             self._platform = PygletPlatform(self.viewport_width,
                                             self.viewport_height)
         elif os.environ['PYOPENGL_PLATFORM'] == 'egl':
-            from pyrender.platforms import egl
+            from .platforms import egl
             device_id = int(os.environ.get('EGL_DEVICE_ID', '0'))
             egl_device = egl.get_device_by_index(device_id)
             self._platform = egl.EGLPlatform(self.viewport_width,
                                              self.viewport_height,
                                              device=egl_device)
         elif os.environ['PYOPENGL_PLATFORM'] == 'osmesa':
-            from pyrender.platforms.osmesa import OSMesaPlatform
+            from .platforms.osmesa import OSMesaPlatform
             self._platform = OSMesaPlatform(self.viewport_width,
                                             self.viewport_height)
         else:
--- a/pyrender/platforms/base.py
+++ b/pyrender/platforms/base.py
@@ -1,10 +1,8 @@
 import abc
 
-import six
 
 
-@six.add_metaclass(abc.ABCMeta)
-class Platform(object):
+class Platform(abc.ABC):
     """Base class for all OpenGL platforms.
 
     Parameters
--- a/pyrender/platforms/pyglet_platform.py
+++ b/pyrender/platforms/pyglet_platform.py
@@ -1,4 +1,4 @@
-from pyrender.constants import (TARGET_OPEN_GL_MAJOR, TARGET_OPEN_GL_MINOR,
+from ..constants import (TARGET_OPEN_GL_MAJOR, TARGET_OPEN_GL_MINOR,
                                 MIN_OPEN_GL_MAJOR, MIN_OPEN_GL_MINOR)
 from .base import Platform
 
@@ -45,21 +45,22 @@
                                   double_buffer=True,
                                   major_version=MIN_OPEN_GL_MAJOR,
                                   minor_version=MIN_OPEN_GL_MINOR)]
+        error = None
         for conf in confs:
             try:
                 self._window = pyglet.window.Window(config=conf, visible=False,
                                                     resizable=False,
                                                     width=1, height=1)
                 break
-            except pyglet.window.NoSuchConfigException as e:
-                pass
+            except pyglet.window.NoSuchConfigException as exc:
+                error = exc
 
         if not self._window:
             raise ValueError(
                 'Failed to initialize Pyglet window with an OpenGL >= 3+ '
                 'context. If you\'re logged in via SSH, ensure that you\'re '
                 'running your script with vglrun (i.e. VirtualGL). The '
-                'internal error message was "{}"'.format(e)
+                'internal error message was "{}"'.format(error)
             )
 
     def make_current(self):
--- a/pyrender/renderer.py
+++ b/pyrender/renderer.py
@@ -13,7 +13,6 @@
 from .shader_program import ShaderProgramCache
 from .material import MetallicRoughnessMaterial, SpecularGlossinessMaterial
 from .light import PointLight, SpotLight, DirectionalLight
-from .font import FontCache
 from .utils import format_color_vector
 
 from OpenGL.GL import *
@@ -61,7 +60,6 @@
 
         # Shader Program Cache
         self._program_cache = ShaderProgramCache()
-        self._font_cache = FontCache()
         self._meshes = set()
         self._mesh_textures = set()
         self._shadow_textures = set()
@@ -185,38 +183,9 @@
             :attr:`TextAlign.BOTTOM_LEFT` means that ``x`` and ``y`` indicate
             the position of the bottom-left corner of the textbox.
         """
-        x *= self.dpscale
-        y *= self.dpscale
-        font_pt *= self.dpscale
-
-        if color is None:
-            color = np.array([0.0, 0.0, 0.0, 1.0])
-        else:
-            color = format_color_vector(color, 4)
-
-        # Set up viewport for render
-        self._configure_forward_pass_viewport(0)
-
-        # Load font
-        font = self._font_cache.get_font(font_name, font_pt)
-        if not font._in_context():
-            font._add_to_context()
-
-        # Load program
-        program = self._get_text_program()
-        program._bind()
-
-        # Set uniforms
-        p = np.eye(4)
-        p[0,0] = 2.0 / self.viewport_width
-        p[0,3] = -1.0
-        p[1,1] = 2.0 / self.viewport_height
-        p[1,3] = -1.0
-        program.set_uniform('projection', p)
-        program.set_uniform('text_color', color)
-
-        # Draw text
-        font.render_string(text, x, y, scale, align)
+        raise NotImplementedError(
+            'Text rendering is not included in this vendored copy of pyrender.'
+        )
 
     def read_color_buf(self):
         """Read and return the current viewport's color buffer.
@@ -287,9 +256,6 @@
         """
         # Free shaders
         self._program_cache.clear()
-
-        # Free fonts
-        self._font_cache.clear()
 
         # Free meshes
         for mesh in self._meshes:
deleted: pyrender/trackball.py
--- a/pyrender/version.py
+++ b/pyrender/version.py
@@ -1 +1 @@
-__version__ = '0.1.45'
+__version__ = '0.1.45+a59963e.smpl_avatar'
deleted: pyrender/viewer.py
